crypto/fipsmodule/service_indicator/service_indicator.c

```diff
@@ -260,14 +260,12 @@ static void evp_md_ctx_verify_service_indicator(const EVP_MD_CTX *ctx,
       }
     }
 
-    // The approved RSA key sizes for signing are 2048, 3072 and 4096 bits.
-    // Note: |EVP_PKEY_size| returns the size in bytes.
-    size_t pkey_size = EVP_PKEY_size(ctx->pctx->pkey);
+    // The approved RSA key sizes for signing are key sizes >= 2048 bits and bits % 2 == 0.
+    size_t n_bits = RSA_bits(ctx->pctx->pkey->pkey.rsa);
 
     // Check if the MD type and the RSA key size are approved.
     if (md_ok(md_type, pkey_type) &&
-        ((rsa_1024_ok && pkey_size == 128) || pkey_size == 256 ||
-         pkey_size == 384 || pkey_size == 512)) {
+        ((rsa_1024_ok && n_bits == 1024) || (n_bits >= 2048 && n_bits % 2 == 0))) {
```

Comment:
I don't think it makes sense in this case as it's not necessarily a key that was generated by AWS-LC, so they key generation constraint of `% 128` wouldn't be applicable no?

Reply:
It's a good point. I then thought we can tie the check to `rsa_1024_ok` being 0 since that would mean the key is used for signing and is generated by aws-lc, but then I guess it can be imported. So we can leave it as is.